tests/module/memory_test.c

```diff
@@ -0,0 +1,191 @@
+
+#include "VecSim/memory/redismodule.h"
+#include "VecSim/vec_sim.h"
+#include <stdbool.h>
+#include <errno.h>
+#include <string.h>
+
+#define DIMENSION 100
+
+// get memory usage from redis-server
+long long _get_memory_usage(RedisModuleCtx *ctx) {
+    long long usage;
+    RedisModuleServerInfoData *info = RedisModule_GetServerInfo(ctx, "memory");
+    RedisModuleString *memory = RedisModule_ServerInfoGetField(ctx, info, "used_memory");
+    RedisModule_StringToLongLong(memory, &usage);
+    RedisModule_FreeServerInfo(ctx, info);
+    RedisModule_FreeString(ctx, memory);
+    return usage;
+}
+
+// adds 'amount' vectors to the index. could be 0
+void _add_vectors(VecSimIndex *index, long long amount) {
+    VecSimIndexInfo indexInfo = VecSimIndex_Info(index);
+    int i;
+    int64_t vec[indexInfo.d];
+    for (i = 0; i < indexInfo.d; i++)
+        vec[i] = i;
+    for (i = 0; i < amount; i++)
```

Comment:
for the sake of consistency in our code, please don't re-use the iteration variables in different scope. 
have `for(int i=0; i < n; i++)`, unless you really must use shared variables